fix(geospatial): Align parameter with server expectation

ngoiyaeric opened the pull request:
### **User description**
A geospatial request was sending a `query` parameter, but the server expected `searchText`. This caused a validation error.

This commit renames the parameter to `searchText` to match the server's API, resolving the error.


___

### **PR Type**
Bug fix


___

### **Description**
- Fix parameter name mismatch in geospatial tool

- Change `query` to `searchText` for server compatibility


___

### Diagram Walkthrough


```mermaid
flowchart LR
  A["Geospatial Tool"] -- "rename parameter" --> B["Server API"]
  A -- "query → searchText" --> B
```



<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Bug fix</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>geospatial.tsx</strong><dd><code>Fix parameter name for server compatibility</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

lib/agents/tools/geospatial.tsx

<ul><li>Rename <code>query</code> parameter to <code>searchText</code> in toolArgs object<br> <li> Maintain same functionality with corrected parameter name</ul>


</details>


  </td>
  <td><a href="https://github.com/QueueLab/QCX/pull/219/files#diff-a1c54448cae8824b75605378f82ffb3169fbb89c33065586d737ed7df2ccb407">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

</details>

___



<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * More reliable geospatial queries with automatic retries, extended timeouts, and clearer progress feedback.
  * Enhanced result parsing and formatting for place names, addresses, and map links; improved map preview handling.
  * UI updates that surface mapping service availability and detailed error information.

* **Bug Fixes**
  * Fixed failures from invalid/missing configuration and ensured connections are properly closed on errors.

* **Chores**
  * Environment validation, masked logging, and general resilience improvements for safer, more predictable behavior.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

google-labs-jules[bot] committed:
fix(geospatial): Align parameter with server expectation

A geospatial request was sending a `query` parameter, but the server expected `searchText`. This caused a validation error.

This commit renames the parameter to `searchText` to match the server's API, resolving the error.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/QueueLab/QCX?pullRequest=219) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/QueueLab/QCX?pullRequest=219) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/QueueLab/QCX?pullRequest=219) it.</sub>

google-labs-jules[bot] committed:
fix(geospatial): Correct server URL and align tool parameter

This commit addresses two issues:

1.  **Parameter Mismatch:** The geospatial tool was sending a `query` parameter, but the server expected `searchText`. This has been corrected.

2.  **Incorrect Server URL:** The tool was constructing an incorrect URL, leading to a 404 error. The `mcpServerBaseUrl` has been updated to the correct endpoint you provided, which includes the necessary user scope.

google-labs-jules[bot] committed:
I've made a commit to fix the geospatial tool. Here's a summary of the changes:

1.  **Authentication:** I've updated the HTTP transport to send the API key as a Bearer token in the `Authorization` header, which resolves the 401 Unauthorized error.
2.  **Server URL:** I've corrected the base URL for the Smithery server to point to your user-specific endpoint, fixing the 404 Not Found error.
3.  **Tool Parameter:** I've changed the `mapbox_geocoding` tool call to use the `searchText` parameter instead of `query`. This aligns with the server's API and fixes the initial tool execution failure.

ngoiyaeric committed:
Update geospatial.tsx

ngoiyaeric committed:
Update geospatial.tsx

ngoiyaeric committed:
Update geospatial.tsx

ngoiyaeric committed:
Update geospatial.tsx

ngoiyaeric committed:
Update geospatial.tsx
